service/controller/clusterapi/v17/resources/clusterid/create.go

```diff
@@ -2,8 +2,55 @@ package clusterid
 
 import (
 	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v17/key"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring cluster status has cluster ID")
+
+	status := r.commonClusterStatusAccessor.GetCommonClusterStatus(cr)
+
+	if status.ID != "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "ensured cluster status has cluster ID")
```

Comment:
IMO this should be "not ensuring ...", since the process of ensuring happened earlier. Here we are not doing anything because the ID is already there. The logs could be clearer with this.

Reply:
I disagree. Ensure, i.e. "make sure", "guarantee", that cluster ID is present in status. I think that happened there. It made sure that it's there and there was no need to go further.